sda/conode.go

```diff
@@ -34,18 +34,10 @@ type Conode struct {
 	protocols *protocolStorage
 }
 
-// NewConode returns a new Host that out of a private-key and its relating public
-// key within the ServerIdentity. The host will create a default TcpRouter as Router.
-func NewConode(e *network.ServerIdentity, pkey abstract.Scalar) *Conode {
-	r, err := network.NewTCPRouter(e)
-	log.ErrFatal(err)
-	return NewConodeWithRouter(e, pkey, r)
-}
-
-// NewConodeWithRouter returns a fresh Host with a given Router.
-func NewConodeWithRouter(e *network.ServerIdentity, pkey abstract.Scalar, r *network.Router) *Conode {
+// NewConode returns a fresh Host with a given Router.
+func NewConode(r *network.Router, pkey abstract.Scalar) *Conode {
 	c := &Conode{
-		ServerIdentity:       e,
+		ServerIdentity:       r.ServerIdentity,
```

Comment:
since now `Router` has an `ServerIdentity` field and is embedded into `Conode` I guess you don't need that field anymore, calling `conode.ServerIdentity` should work.